disdrodb/data_transfer/download_data.py

```diff
@@ -0,0 +1,164 @@
+    # Get the list of url and local paths from the config files
+    list_of_path_url = []
+    for yaml_file_path in yaml_files:
+        station_dir_path, data_url = get_station_local_remote_locations(yaml_file_path)
+        if data_url is None:
+            continue
+        list_of_path_url.append((station_dir_path, data_url))
+
+    return list_of_path_url
+
+
+def download_file_from_url(url: str, dir_path: str) -> None:
+    """Download file.
+
+    Parameters
+    ----------
+
+    url : str
+        URL of the file to download.
+    file_name : str
+        Local name of the file.
+    """
+
+    fname = os.path.basename(url)
+
+    downloader = pooch.HTTPDownloader(progressbar=True)
+    pooch.retrieve(
+        url=url,
+        known_hash=None,
+        path=dir_path,
+        fname=fname,
+        downloader=downloader,
+        progressbar=tqdm,
+    )
+
+
+def download_single_archive(url_local_path: tuple, overwrite: bool = False) -> None:
+    """Download the files based on a tuple (local path, url).
+
+    Parameters
+    ----------
+    file_path : tuple
+        Tuple containing the local file path and the url.
+    overwrite : bool, optional
+        Overwrite the raw data file if already existing, by default False.
+
+    """
+
+    local_path, url = url_local_path
+    if check_url(url):
+        # create station folder if not existing
+        if not os.path.exists(local_path):
+            os.makedirs(local_path)
+        # download the file
+        download_file_from_url(url, local_path)
+        print(f"Download {url} into {local_path}")
+
+    else:
+        # raise an error if url not well formatted
+        raise ValueError(f"URL {url} is not well formatted. Please check the URL.")
+
+
+def download_disdrodb_archives(
+    disdrodb_dir: str = None,
+    data_source: str = None,
+    campaign_name: str = None,
+    station_name: str = None,
+    overwrite: bool = False,
+):
+    """Batch function to get all YAML files that contain
+    the 'data_url' key and download the data locally.
+
+    Parameters
+    ----------
+    disdrodb_dir : str, optional
+        DisdroDB data folder path.
+        Must end with DISDRODB/Raw.
+    data_source : str, optional
+        Data source folder name (eg : EPFL).
+        If not provided (None), all data sources will be downloaded.
+        The default is data_source=None.
+    campaign_name : str, optional
+        Campaign name (eg :  EPFL_ROOF_2012).
+        If not provided (None), all campaigns will be downloaded.
+        The default is campaign_name=None.
+    station_name : str, optional
+        Station name.
+        If not provided (None), all stations will be downloaded.
+        The default is station_name=None.
+    overwrite : bool, optional
+        If True, overwrite the already existing raw data file.
+        The default is False.
+    """
+
+    list_urls_data_dir = _get_local_and_remote_data_directories(disdrodb_dir, data_source, campaign_name, station_name)

```

Comment:
This could be defined as : 
```
# Get requested metadata file paths 
yaml_files = get_metadata_list(disdrodb_dir=disdrodb_dir, data_sources=data_sources, campaign_names=campaign_names, station_names=station_names)
# Loop over YAML metadata files
 for yaml_file_path in yaml_files:
        # Retrieve local station directory and remote station url 
        station_dir_path, data_url = _get_station_local_remote_locations(yaml_file_path)
        # Download station data 
        _download_station_data(station_dir_path, data_url) 
 ```
 Just edit `_download_station_data` for input arguments and do nothing if `data_url is None`